diff -ru gps-release-ide-5.2.1-src_compilebase/common/src/gui_utils.adb gps-release-ide-5.2.1-src/common/src/gui_utils.adb
--- gps-release-ide-5.2.1-src_compilebase/common/src/gui_utils.adb	2011-12-20 10:58:07.000000000 +0100
+++ gps-release-ide-5.2.1-src/common/src/gui_utils.adb	2013-06-30 16:33:09.000000000 +0200
@@ -1038,7 +1038,9 @@
       Shift   : constant String := "shift-";
       Meta    : constant String := "alt-";
       Control : constant String := "control-";
-      Max : constant Natural := Shift'Length + Control'Length + Meta'Length;
+      Command : constant String := "command-";
+      Max : constant Natural := Shift'Length + Control'Length +
+                                Meta'Length + Command'Length;
       Buffer   : String (1 .. Max);
       Current : Natural := Buffer'First;
 
@@ -1077,6 +1079,11 @@
                Current := Current + Meta'Length;
             end if;
 
+            if (Mods and Meta_Mask) /= 0 then
+               Buffer (Current .. Current + Command'Length - 1) := Command;
+               Current := Current + Command'Length;
+            end if;
+
             return
               Buffer (Buffer'First .. Current - 1) & Gdk.Keyval.Name (Key);
       end case;
@@ -1102,6 +1109,8 @@
                Mods := Mods or Control_Mask;
             elsif From (Start .. D) = "alt-" then
                Mods := Mods or Mod1_Mask;
+            elsif From (Start .. D) = "command-" then
+               Mods := Mods or Meta_Mask;
             end if;
             Start := D + 1;
          end if;
diff -ru gps-release-ide-5.2.1-src_compilebase/keymanager/src/keymanager_module.adb gps-release-ide-5.2.1-src/keymanager/src/keymanager_module.adb
--- gps-release-ide-5.2.1-src_compilebase/keymanager/src/keymanager_module.adb	2012-01-31 12:25:23.000000000 +0100
+++ gps-release-ide-5.2.1-src/keymanager/src/keymanager_module.adb	2013-06-30 16:32:41.000000000 +0200
@@ -853,7 +853,8 @@
                   if Modif = Mnemonic_Modif then
                      Set_Mnemonic_Modifier
                        (Get_Main_Window (Kernel),
-                        Modif or Control_Mask or Mod1_Mask or Shift_Mask);
+                        Modif or Control_Mask or Mod1_Mask or Shift_Mask
+                        or Meta_Mask);
                   end if;
 
                   if not Change_Entry
